lib/config/configuration.go

```diff
@@ -809,6 +807,24 @@ func applyProxyConfig(fc *FileConfig, cfg *service.Config) error {
 	return nil
 }
 
+func getPostgresDefaultPort(fc *FileConfig, cfg *service.Config) int {
+	if fc.Proxy.PostgresAddr != "" {
```

Comment:
Also, why is this checking `fc` (file config) but returning the value from `cfg` (service config)?

Reply:
> Also, why is this checking fc (file config) but returning the value from cfg (service config)?

Fixed. 

> Shouldn't the port from Postgres public address (if it's specified) take precedence?

In case of proxy configuration no, the `PostgresAddr` is a local proxy listener address and in case of local proxy listener the PublicAddress port should not take precedence over local proxy listener port. 

though `PostgresPublicAddr` should take precedence over local port in case of Proxy Ping response but for this a separate function is used.